File: instedit/pas/reset_marked_instruments.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure reset_marked_instruments;
begin
  If (marked_instruments < 2) then
   If (songdata.instr_names[instrum_page][1] = _ins_mark_chr) then
     songdata.instr_names[instrum_page][1] := ' '
   else
  else begin
         songdata.instr_names[_2nd_marked][1] := ' ';
         songdata.instr_names[_1st_marked][1] := ' ';
       end;
  update_4op_flag_marks;
end;

File: instedit/reset_marked_instruments.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2025 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void reset_marked_instruments (void)
{
  if (marked_instruments () < 2)
  {
    unsigned char *c;

    c = &(GetStr (songdata.instr_names[instrum_page - 1])[0]);
    if (*c == _INS_MARK_CHR)
      *c = ' ';
  }
  else
  {
    GetStr (songdata.instr_names[_2nd_marked () - 1])[0] = ' ';
    GetStr (songdata.instr_names[_1st_marked () - 1])[0] = ' ';
  }
  update_4op_flag_marks ();
}
